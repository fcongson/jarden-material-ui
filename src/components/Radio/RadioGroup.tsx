import { RadioGroup as MaterialRadioGroup, ThemeProvider } from '@material-ui/core'
import * as React from 'react'
import { theme } from '../../styles/material'

interface RadioGroupProps {
  name?: string
  value?: any
  onChange: (value: any) => void
}

export const RadioGroup: React.FC<RadioGroupProps> = ({ name, value, onChange, children, ...rest }) => (
  <ThemeProvider theme={theme}>
    <MaterialRadioGroup
      name={name}
      value={value}
      onChange={event => {
        if (!!onChange) onChange(event.target.value)
      }}
      {...rest}>
      {children}
    </MaterialRadioGroup>
  </ThemeProvider>
)
