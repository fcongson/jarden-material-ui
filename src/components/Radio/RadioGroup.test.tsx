import { shallow } from 'enzyme'
import * as React from 'react'
import { RadioGroup } from './RadioGroup'

describe('RadioGroup', () => {
  it('Renders', () => {
    const radioGroup = shallow(<RadioGroup onChange={() => {}} />)
    expect(radioGroup).toMatchSnapshot()
  })
})
