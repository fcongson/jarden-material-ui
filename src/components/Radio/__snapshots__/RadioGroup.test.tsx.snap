// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup Renders 1`] = `ShallowWrapper {}`;
